ViAn/Analysis/MotionDetection.cpp

```diff
@@ -0,0 +1,78 @@
+#include "MotionDetection.h"
+#include <vector>
+
+/**
+ * @brief MotionDetection::MotionDetection
+ * @param source_file
+ */
+MotionDetection::MotionDetection(std::string source_file) {
+    m_analysis.type = MOTION_DETECTION;
+    capture.open(source_file);
+}
+
+/**
+ * @brief MotionDetection::setup_analysis
+ * Initial setup of the analysis
+ */
+void MotionDetection::setup_analysis(){
+    pMOG2 = cv::createBackgroundSubtractorMOG2(500,50,false);
+}
+
+/**
+ * @brief MotionDetection::do_analysis
+ * Motion detection specific code
+ * This function is called from the interface thread loop
+ */
+std::vector<OOI> MotionDetection::analyse_frame(){

```

Comment:
this method seems rather complex, if it should be maintainable, perhaps som clarifying comments about implementation is required